### PR TITLE
Enable targetting AOTC Android builds from Mac hosts.

### PR DESCRIPTION
**Pull request by chinmaygarde**
cc @yjbanov, @jason-simmons, @sethladd 

Will land once the bots settle down at 053630dc092ff21fbab53dbc474a832e7484eb67

### CODE REVIEW

**Comment by yjbanov** [comment]
LGTM

File: packages/flutter_tools/lib/src/cache.dart [hunk 1]
```diff
@@ -220,7 +220,11 @@ class FlutterEngine {
   // Return a list of (cache directory path, download URL path) tuples.
   List<List<String>> _getToolsDirs() {
     if (Platform.isMacOS)
-      return <List<String>>[<String>['mac_debug', 'mac_debug/artifacts.zip']];
+      return <List<String>>[
+        <String>['darwin-x64', 'darwin-x64/artifacts.zip'],
+        <String>['android-arm-profile/darwin-x64', 'android-arm-profile/darwin-x64.zip'],
```
Comment (eseidelGoogle) [inline packages/flutter_tools/lib/src/cache.dart]: Confused as to why we have this additional layer of mapping from bucket names to on-disk names.  But it's OK too.

File: packages/flutter_tools/lib/src/build_info.dart [hunk 2]
```diff
@@ -62,7 +62,7 @@ String getNameForTargetPlatform(TargetPlatform platform) {
     case TargetPlatform.ios:
       return "ios_release";
     case TargetPlatform.darwin_x64:
-      return "mac_debug";
+      return "darwin-x64";
```
Comment (eseidelGoogle) [inline packages/flutter_tools/lib/src/build_info.dart]: Do we use ' or " for quotes in Dart?

Comment (abarth) [inline packages/flutter_tools/lib/src/build_info.dart]: We use single quotes.

**Comment by chinmaygarde** [comment]
Done & Rebased. Update the revision cooking on the bot. Its not done yet. Will commit once the tests pass.

> Confused as to why we have this additional layer of mapping from bucket names to on-disk names. 

Just replicated linux paths below. Will clean it up in a subsequent patch.

**Comment by chinmaygarde** [comment]
Bots done and travis is green. Landing.

**Pull request merged**